### DISCUSSION TITLE
aws-iot-device-sdk-python-v2 build fails (kirkstone)

### DISCUSSION THREAD

**Discussion by eflavio** [aws4embeddedlinux/meta-aws#4475] [Q&A] [answered]
Hi,

I want to build aws-iot-device-sdk-python-v2 in yocto kirkstone. However, the build fails at the dependency aws-c-sdkutils. 
This is the bitbake output when I want to build aws-iot-device-sdk-python-v2:

`
Build Configuration:
BB_VERSION           = "2.0.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "arm-phytec-linux-gnueabi"
MACHINE              = "ksp0671-1"
DISTRO               = "ampliphy"
DISTRO_VERSION       = "BSP-Yocto-Ampliphy-i.MX6-PD22.1.0"
TUNE_FEATURES        = "arm vfp cortexa9 neon thumb callconvention-hard"
TARGET_FPU           = "hard"
workspace            = "kirkstone:ac518546dc60b7e74bfe3b4234e6af52daecda93"
meta                 
meta-poky            = "HEAD:c8423297866441055553b378663b5454e63e18b4"
meta-oe              
meta-networking      
meta-python          
meta-multimedia      
meta-filesystems     
meta-perl            = "HEAD:50d4a8d2a983a68383ef1ffec2c8e21adf0c1a79"
meta-qt5             = "HEAD:44d44933200287f7d17cf6981af4b4a0961c308d"
meta-rauc            = "HEAD:e2a60ded9c08118c17c23aa07cca21df77fb6628"
meta-security        
meta-tpm             = "HEAD:c79262a30bd385f5dbb009ef8704a1a01644528e"
meta-qt5-phytec      = "HEAD:bd9ac458bac01b899308ac173444607a10f6cc43"
meta-ampliphy        = "HEAD:3c37615e96fb5fe0230abc6a09773e7b89250179"
meta-phytec          = "HEAD:ffcd0a44c8868b9af645e6122ff3ddfb53b2db0f"
meta-rqm             = "HEAD:4a7d79c71ed8a3fffc59b470e73c5c5d75c85fcb"
meta-ksp0671         = "HEAD:2c60d05394016264c176dffc6ff764efa2c269da"
meta-aws             = "kirkstone-next:4c2b6ddaf880f1025a6a7b6d77e0b7647d86b8d2"
meta-iot-cloud       = "kirkstone:65e2c5b2e5cb33760b2deb0fc249ab43ebdf08ef"

Initialising tasks: 100% |##################################################################################################################################################################| Time: 0:00:03
Sstate summary: Wanted 62 Local 9 Mirrors 0 Missed 53 Current 2819 (14% match, 98% complete)
NOTE: Executing Tasks
ERROR: aws-c-sdkutils-0.1.11-r0 do_compile: ExecutionError('/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/temp/run.do_compile.425191', 1, None, None)
ERROR: Logfile of failure stored in: /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/temp/log.do_compile.425191
Log data follows:
| DEBUG: Executing shell function do_compile
| NOTE: VERBOSE=1 cmake --build /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/build --target all --
| [1/11] /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native/usr/bin/arm-phytec-linux-gnueabi/arm-phytec-linux-gnueabi-gcc -DAWS_SDKUTILS_EXPORTS -DAWS_SDKUTILS_USE_IMPORT_EXPORT -DHAVE_SYSCONF -Daws_c_sdkutils_EXPORTS -I/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native=  -Wl,-Bsymbolic -DNDEBUG -fPIC -Wall -Werror -Wstrict-prototypes -fno-omit-frame-pointer -pedantic -Wno-long-long -fPIC -fvisibility=hidden -std=gnu99 -MD -MT CMakeFiles/aws-c-sdkutils.dir/source/endpoints_standard_lib.c.o -MF CMakeFiles/aws-c-sdkutils.dir/source/endpoints_standard_lib.c.o.d -o CMakeFiles/aws-c-sdkutils.dir/source/endpoints_standard_lib.c.o -c /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_standard_lib.c
| FAILED: CMakeFiles/aws-c-sdkutils.dir/source/endpoints_standard_lib.c.o
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native/usr/bin/arm-phytec-linux-gnueabi/arm-phytec-linux-gnueabi-gcc -DAWS_SDKUTILS_EXPORTS -DAWS_SDKUTILS_USE_IMPORT_EXPORT -DHAVE_SYSCONF -Daws_c_sdkutils_EXPORTS -I/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native=  -Wl,-Bsymbolic -DNDEBUG -fPIC -Wall -Werror -Wstrict-prototypes -fno-omit-frame-pointer -pedantic -Wno-long-long -fPIC -fvisibility=hidden -std=gnu99 -MD -MT CMakeFiles/aws-c-sdkutils.dir/source/endpoints_standard_lib.c.o -MF CMakeFiles/aws-c-sdkutils.dir/source/endpoints_standard_lib.c.o.d -o CMakeFiles/aws-c-sdkutils.dir/source/endpoints_standard_lib.c.o -c /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_standard_lib.c
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_standard_lib.c:6:10: fatal error: aws/common/json.h: No such file or directory
|     6 | #include <aws/common/json.h>
|       |          ^~~~~~~~~~~~~~~~~~~
| compilation terminated.
| [2/11] /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native/usr/bin/arm-phytec-linux-gnueabi/arm-phytec-linux-gnueabi-gcc -DAWS_SDKUTILS_EXPORTS -DAWS_SDKUTILS_USE_IMPORT_EXPORT -DHAVE_SYSCONF -Daws_c_sdkutils_EXPORTS -I/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native=  -Wl,-Bsymbolic -DNDEBUG -fPIC -Wall -Werror -Wstrict-prototypes -fno-omit-frame-pointer -pedantic -Wno-long-long -fPIC -fvisibility=hidden -std=gnu99 -MD -MT CMakeFiles/aws-c-sdkutils.dir/source/endpoints_rule_engine.c.o -MF CMakeFiles/aws-c-sdkutils.dir/source/endpoints_rule_engine.c.o.d -o CMakeFiles/aws-c-sdkutils.dir/source/endpoints_rule_engine.c.o -c /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_rule_engine.c
| FAILED: CMakeFiles/aws-c-sdkutils.dir/source/endpoints_rule_engine.c.o
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native/usr/bin/arm-phytec-linux-gnueabi/arm-phytec-linux-gnueabi-gcc -DAWS_SDKUTILS_EXPORTS -DAWS_SDKUTILS_USE_IMPORT_EXPORT -DHAVE_SYSCONF -Daws_c_sdkutils_EXPORTS -I/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native=  -Wl,-Bsymbolic -DNDEBUG -fPIC -Wall -Werror -Wstrict-prototypes -fno-omit-frame-pointer -pedantic -Wno-long-long -fPIC -fvisibility=hidden -std=gnu99 -MD -MT CMakeFiles/aws-c-sdkutils.dir/source/endpoints_rule_engine.c.o -MF CMakeFiles/aws-c-sdkutils.dir/source/endpoints_rule_engine.c.o.d -o CMakeFiles/aws-c-sdkutils.dir/source/endpoints_rule_engine.c.o -c /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_rule_engine.c
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_rule_engine.c:7:10: fatal error: aws/common/json.h: No such file or directory
|     7 | #include <aws/common/json.h>
|       |          ^~~~~~~~~~~~~~~~~~~
| compilation terminated.
| [3/11] /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native/usr/bin/arm-phytec-linux-gnueabi/arm-phytec-linux-gnueabi-gcc -DAWS_SDKUTILS_EXPORTS -DAWS_SDKUTILS_USE_IMPORT_EXPORT -DHAVE_SYSCONF -Daws_c_sdkutils_EXPORTS -I/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native=  -Wl,-Bsymbolic -DNDEBUG -fPIC -Wall -Werror -Wstrict-prototypes -fno-omit-frame-pointer -pedantic -Wno-long-long -fPIC -fvisibility=hidden -std=gnu99 -MD -MT CMakeFiles/aws-c-sdkutils.dir/source/endpoints_ruleset.c.o -MF CMakeFiles/aws-c-sdkutils.dir/source/endpoints_ruleset.c.o.d -o CMakeFiles/aws-c-sdkutils.dir/source/endpoints_ruleset.c.o -c /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_ruleset.c
| FAILED: CMakeFiles/aws-c-sdkutils.dir/source/endpoints_ruleset.c.o
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native/usr/bin/arm-phytec-linux-gnueabi/arm-phytec-linux-gnueabi-gcc -DAWS_SDKUTILS_EXPORTS -DAWS_SDKUTILS_USE_IMPORT_EXPORT -DHAVE_SYSCONF -Daws_c_sdkutils_EXPORTS -I/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native=  -Wl,-Bsymbolic -DNDEBUG -fPIC -Wall -Werror -Wstrict-prototypes -fno-omit-frame-pointer -pedantic -Wno-long-long -fPIC -fvisibility=hidden -std=gnu99 -MD -MT CMakeFiles/aws-c-sdkutils.dir/source/endpoints_ruleset.c.o -MF CMakeFiles/aws-c-sdkutils.dir/source/endpoints_ruleset.c.o.d -o CMakeFiles/aws-c-sdkutils.dir/source/endpoints_ruleset.c.o -c /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_ruleset.c
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/endpoints_ruleset.c:8:10: fatal error: aws/common/json.h: No such file or directory
|     8 | #include <aws/common/json.h>
|       |          ^~~~~~~~~~~~~~~~~~~
| compilation terminated.
| [4/11] /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native/usr/bin/arm-phytec-linux-gnueabi/arm-phytec-linux-gnueabi-gcc -DAWS_SDKUTILS_EXPORTS -DAWS_SDKUTILS_USE_IMPORT_EXPORT -DHAVE_SYSCONF -Daws_c_sdkutils_EXPORTS -I/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native=  -Wl,-Bsymbolic -DNDEBUG -fPIC -Wall -Werror -Wstrict-prototypes -fno-omit-frame-pointer -pedantic -Wno-long-long -fPIC -fvisibility=hidden -std=gnu99 -MD -MT CMakeFiles/aws-c-sdkutils.dir/source/aws_profile.c.o -MF CMakeFiles/aws-c-sdkutils.dir/source/aws_profile.c.o.d -o CMakeFiles/aws-c-sdkutils.dir/source/aws_profile.c.o -c /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c
| FAILED: CMakeFiles/aws-c-sdkutils.dir/source/aws_profile.c.o
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native/usr/bin/arm-phytec-linux-gnueabi/arm-phytec-linux-gnueabi-gcc -DAWS_SDKUTILS_EXPORTS -DAWS_SDKUTILS_USE_IMPORT_EXPORT -DHAVE_SYSCONF -Daws_c_sdkutils_EXPORTS -I/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include -mthumb -mfpu=neon -mfloat-abi=hard -mcpu=cortex-a9 -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security  --sysroot=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot  -O2 -pipe -g -feliminate-unused-debug-types -fmacro-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0=/usr/src/debug/aws-c-sdkutils/0.1.11-r0                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot=                      -fdebug-prefix-map=/home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/recipe-sysroot-native=  -Wl,-Bsymbolic -DNDEBUG -fPIC -Wall -Werror -Wstrict-prototypes -fno-omit-frame-pointer -pedantic -Wno-long-long -fPIC -fvisibility=hidden -std=gnu99 -MD -MT CMakeFiles/aws-c-sdkutils.dir/source/aws_profile.c.o -MF CMakeFiles/aws-c-sdkutils.dir/source/aws_profile.c.o.d -o CMakeFiles/aws-c-sdkutils.dir/source/aws_profile.c.o -c /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c
| In file included from /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include/aws/sdkutils/aws_profile.h:8,
|                  from /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c:13:
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include/aws/sdkutils/sdkutils.h:13:28: error: expected ';' before 'struct'
|    13 | AWS_PUSH_SANE_WARNING_LEVEL
|       |                            ^
|       |                            ;
|    14 |
|    15 | struct aws_allocator;
|       | ~~~~~~
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include/aws/sdkutils/sdkutils.h:52:1: error: unknown type name 'AWS_POP_SANE_WARNING_LEVEL'
|    52 | AWS_POP_SANE_WARNING_LEVEL
|       | ^~~~~~~~~~~~~~~~~~~~~~~~~~
| In file included from /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c:13:
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include/aws/sdkutils/aws_profile.h:10:28: error: expected ';' before 'struct'
|    10 | AWS_PUSH_SANE_WARNING_LEVEL
|       |                            ^
|       |                            ;
|    11 |
|    12 | struct aws_allocator;
|       | ~~~~~~
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/include/aws/sdkutils/aws_profile.h:219:27: error: expected ';' before 'struct'
|   219 | AWS_POP_SANE_WARNING_LEVEL
|       |                           ^
|       |                           ;
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c: In function 'aws_profile_collection_new_from_file':
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c:1318:9: error: implicit declaration of function 'aws_byte_buf_init_from_file'; did you mean 'aws_byte_buf_init_copy'? [-Werror=implicit-function-declaration]
|  1318 |     if (aws_byte_buf_init_from_file(&file_contents, allocator, aws_string_c_str(file_path)) != 0) {
|       |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
|       |         aws_byte_buf_init_copy
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c: In function 's_process_profile_file_path':
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c:1355:37: error: implicit declaration of function 'aws_get_platform_directory_separator' [-Werror=implicit-function-declaration]
|  1355 |     char local_platform_separator = aws_get_platform_directory_separator();
|       |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c:1363:13: error: implicit declaration of function 'aws_is_any_directory_separator' [-Werror=implicit-function-declaration]
|  1363 |         if (aws_is_any_directory_separator(value)) {
|       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c:1404:34: error: implicit declaration of function 'aws_get_home_directory' [-Werror=implicit-function-declaration]
|  1404 |                 home_directory = aws_get_home_directory(allocator);
|       |                                  ^~~~~~~~~~~~~~~~~~~~~~
| /home/rqm/projects/yocto_hardknott/build/tmp/work/cortexa9t2hf-neon-phytec-linux-gnueabi/aws-c-sdkutils/0.1.11-r0/git/source/aws_profile.c:1404:32: error: assignment to 'struct aws_string *' from 'int' makes pointer from integer without a cast [-Werror=int-conversion]
|  1404 |                 home_directory = aws_get_home_directory(allocator);
|       |                                ^
| cc1: all warnings being treated as errors
| ninja: build stopped: subcommand failed.
| WARNING: exit code 1 from a shell command.
ERROR: Task (/home/rqm/projects/yocto_hardknott/sources/poky/../meta-aws/recipes-sdk/aws-c-sdkutils/aws-c-sdkutils_0.1.11.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 3682 tasks of which 3681 didn't need to be rerun and 1 failed.
NOTE: Writing buildhistory
NOTE: Writing buildhistory took: 3 seconds

Summary: 1 task failed:
  /home/rqm/projects/yocto_hardknott/sources/poky/../meta-aws/recipes-sdk/aws-c-sdkutils/aws-c-sdkutils_0.1.11.bb:do_compile
Summary: There were 3 WARNING messages.
Summary: There was 1 ERROR message, returning a non-zero exit code.
`

**Comment by thomas-roos** [comment]
Hi, tried to reproduce this error, but works just fine.
Might the issue is in the meta-ksp0671 ?
Look here what I've tried: https://github.com/aws4embeddedlinux/meta-aws-demos/tree/master-next
Might you can provide a minimal example to reproduce this issue based on this?

**Comment by eflavio** [comment, marked as answer]
Hi Thomas,
Thanks for checking. Indeed, the problem lied in one of the other layers, namely meta-iot-cloud. It also contains recipes related to AWS and had higher layer priority than meta-aws.

I lowered the layer priority of meta-iot-cloud to 5 and everything works fine. I achieved this by adding this line to my local.conf file:
`BBFILE_PRIORITY_iot-cloud = "5"`
